backend/pkg/proto/service.go

```diff
@@ -227,26 +244,30 @@ func (s *Service) SerializeJSONToConfluentProtobufMessage(json []byte, schemaID
 		return nil, err
 	}
 
-	msg := dynamic.NewMessage(messageDescriptor)
-	err = msg.UnmarshalJSONPB(&jsonpb.Unmarshaler{
-		AnyResolver:        &anyResolver{s.registry},
-		AllowUnknownFields: true,
-	}, json)
+	// Convert to modern dynamicpb message
+	modernDesc := convertDescToProtoReflect(messageDescriptor)
+	modernMsg := dynamicpb.NewMessage(modernDesc)
+
+	// Use modern protojson to unmarshal
+	err = protojson.UnmarshalOptions{
```

Comment:
NIT: Maybe the "modern" bits are a bit weird in the future for those who don't know what we did before :D